src/olympia/promoted/tests/test_models.py

```diff
@@ -427,6 +427,54 @@ def test_approve_for_addon(self):
         ).exists()
 
 
+class TestPromotedGroupManager(TestCase):
+    def setUp(self):
+        self.addon: Addon = addon_factory()
+        self.promoted_group = PromotedGroup.objects.get(
+            group_id=PROMOTED_GROUP_CHOICES.SPOTLIGHT
+        )
+        self.application_id = applications.FIREFOX.id
+
+    def test_approved_for(self):
```

Comment:
Nit: This test is covering a lot of different behaviors. Both in terms of identifying regressions and maintaining the test going forward, it will benefit from being split to more specific tests for 1 or a small group of very tight behavior.

I can see 3-4 tests here:
- test_unapproved_promoted_addon_promotion
- test_approved_promoted_addon_promotion
- test_new_version_invalidates_approval
- test_promoted_group_non_pre_reviewed

You can clean up the boilerplate easily enough by adding a helper methods or properties in setUp before hand.